### PR TITLE
fabtests/multinode: enable multinode collective tests on verbs provider

### LINKED ISSUE #8059: Multinode tests can only be run with tcp provider
It is impossible to set ``verbs`` provider with ``-p`` argument as it is always overwritten by the ``setup_hints()`` function:
https://github.com/ofiwg/libfabric/blob/9d5395650629e9e192abb1b889b5231de71b2790/fabtests/multinode/src/core_coll.c#L528

### PR DESCRIPTION
**Pull request by grom72**
Fixes #8059

### CODE REVIEW

File: fabtests/multinode/src/core_coll.c [hunk 1]
```diff
@@ -525,7 +525,8 @@ static inline int setup_hints()
 	hints->mode = FI_CONTEXT;
 	hints->domain_attr->control_progress = FI_PROGRESS_MANUAL;
 	hints->domain_attr->data_progress = FI_PROGRESS_MANUAL;
-	hints->fabric_attr->prov_name = strdup("tcp");
+	if (!hints->fabric_attr->prov_name)
```
Comment (ooststep) [inline fabtests/multinode/src/core_coll.c:528]: I wonder if we can just remove this completely and either let the `-p` option set it or let getinfo decide

Comment (shefty) [reply fabtests/multinode/src/core_coll.c:528]: We should remove this and align the behavior with all other tests.

**Comment by shefty** [comment]
Please squash the 2 commits into 1 and ensure the signed-off-by line is set correctly, so the DCO check passes.

**Comment by shefty** [comment]
kicked off CI

**Comment by shefty** [comment]
This is a trivial change that got ignored for a bit.  Re-started CI.

**Review by shefty — APPROVED** [review]

**Comment by shefty** [comment]
Pulled changes into PR #8217, so I could rebase to force new CI runs.

**Pull request closed without merging**